codingStandards/AdditionalGuidelines-Java.md

```diff
@@ -0,0 +1,98 @@
+# Additional Guidelines - Java
+
+## Argument validation
```

Comment:
I couldn't find a good reference for this section, so had to cook up my own guidelines. Do review with a very critical mind.